Podcast/SettingsPageViewController.swift

```diff
@@ -0,0 +1,207 @@
+ * array will dynamically create a settings view with desired actions.
+ */
+class SettingsPageViewController: ViewController, UITableViewDelegate, UITableViewDataSource {
+
+    // View items
+    private var tableView: UITableView!
+    private var saveButton: UIBarButtonItem!
+    
+    // Can change this but not recommended
+    var sectionSpacing: CGFloat = 18
+    
+    // Settable configurations
+    var sections: [SettingsSection] = []
+    var showSave: Bool! {
+        didSet {
+            if showSave {
+                navigationItem.rightBarButtonItem = saveButton
+            } else {
+                navigationItem.rightBarButtonItem = nil
+            }
+        }
+    }
+    var returnOnSave:Bool = true
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        navigationController?.setNavigationBarHidden(false, animated: false)
+        
+        tableView = UITableView(frame: CGRect.zero, style: .grouped)
+        tableView.showsVerticalScrollIndicator = true
+        tableView.allowsSelection = false
+        tableView.backgroundColor = .paleGrey
+        for type in SettingsFieldType.allValues {
+            let (reuseId, cellClass):(String, AnyClass) = type.reuseIdAndClass
+            tableView.register(cellClass, forCellReuseIdentifier: reuseId)
+        }
+        tableView.separatorInset = UIEdgeInsetsMake(0, 0, 0, 0)
+        tableView.dataSource = self
+        tableView.delegate = self
+        tableView.allowsSelection = true
+        mainScrollView = tableView
+        view.addSubview(tableView)
+        
+        saveButton = UIBarButtonItem(barButtonSystemItem: .save, target: self, action: #selector(saveAction))
+        if showSave {
+            navigationItem.rightBarButtonItem = saveButton
+        } else {
+            navigationItem.rightBarButtonItem = nil
+        }
+        
+        tableView.snp.makeConstraints({ make in
+            make.edges.equalToSuperview()
+        })
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        tableView.reloadData()
+    }
+    
+    @objc private func saveAction() {
+        for (i, section) in sections.enumerated() {
+            for (j, setting) in section.items.enumerated() {
+                switch setting.type {
+                case .textField:
+                    guard let cell = tableView.cellForRow(at: IndexPath(row: j, section: i)) as? TextFieldSettingsTableViewCell else {
+                        return
+                    }
+                    setting.saveFunction(cell.getTextInput())
+                    break
+                default: setting.saveFunction(())
+                }
+            }
+        }
+        if returnOnSave {
+            self.navigationController?.popViewController(animated: true)
+        }
+    }
+    
+    func setError(at index: IndexPath, with message: String) {
+        // TODO: need fixing
```

Comment:
what fixing? like display a pop-up?

Reply:
My accessory view wasn't working but I decided to wait on fixing this until I asked eileen about the design for what should happen anyway.